src/node.rs

```diff
@@ -1,215 +1,38 @@
-        Node {
-            pos: Position::Absolute(0.0, 0.0),
-            dim: [64.0, 64.0],
-            depth: 0.0,
-            maybe_label: None,
-            style: Style::new(),
-        }
-    }
-}
-
-impl<'a> CustomWidget for Node<'a> {
-
-    type State = NodeState;
-
-    fn update<C>(self, state: NodeState, ui_id: UiId, ui: &mut Ui<C, Node<'a>>) -> WidgetUpdate<NodeState>
-        //where C: CharacterCache //I had to comment this out, not sure why it wouldn't compile for me
-    {
-
-        use elmesque::form::{collage, rect, text};
-        use conrod::utils::is_over_rect;
-
-        let dim = self.dim;
-        let mut xy = ui.get_xy(self.pos, dim, ui.theme.align.horizontal, ui.theme.align.vertical);
-
-        let mouse = ui.get_mouse_state(ui_id).relative_to(xy);
-        let is_over = is_over_rect([0.0, 0.0], mouse.xy, dim);
-        let new_state = get_new_state(is_over, state, mouse);
-
-        match new_state {
-            NodeState::Dragging => xy = ui.mouse.xy, //this isn't exact, fixme!
-            _ => (),
-        }
-        
-        // TODO - use color, frame attributes from style
-        // TODO - refactor element building into a 'new_element' function/method (see builtin Conrod widgets)
-
-        let frame_form = rect(dim[0], dim[1]).filled(ui.theme.frame_color);
-
-        let color = new_state.color(ui.theme.shape_color);
-
-        let frame_w = ui.theme.frame_width;
-        let (inner_w, inner_h) = (dim[0] - frame_w * 2.0, dim[1] - frame_w * 2.0);
-        let inner_form = rect(inner_w, inner_h).filled(color);
-
-        let maybe_label_form = self.maybe_label.map(|label_text| {
-            use elmesque::text::Text;
-            let text_color = ui.theme.label_color;
-            let size = ui.theme.font_size_medium;
-            text(Text::from_string(label_text.to_string()).color(text_color).height(size as f64))
-                .shift(xy[0].floor(), xy[1].floor())
-        });
-
-        let form_chain = Some(frame_form).into_iter()
-            .chain(Some(inner_form).into_iter())
-            .map(|form| form.shift(xy[0], xy[1]))
-            .chain(maybe_label_form.into_iter());
-
-        let element = collage(dim[0] as i32, dim[1] as i32, form_chain.collect());
-
-        WidgetUpdate {
-            new_state: new_state,
-            xy: xy,
-            depth: self.depth,
-            element: element
-        }
+    pub fn new(pos:[f64;2]) -> Self {
+        NodeState { position:pos }
     }
 }
 
-impl<'a> Labelable<'a> for Node<'a> {
-    fn label(mut self, text: &'a str) -> Self {
-        self.maybe_label = Some(text);
-        self
-    }
-
-    fn label_color(mut self, color: Color) -> Self {
-        self.style.maybe_label_color = Some(color);
-        self
-    }
-
-    fn label_font_size(mut self, size: FontSize) -> Self {
-        self.style.maybe_label_font_size = Some(size);
-        self
-    }
+#[derive(Debug)]
+pub struct Edge;
+impl EditableEdge for Edge {
```

Comment:
My original intent was for this to not be implemented by the library itself, but rather users of the library, as different users might need different data associated with an edge, e.g. a 'weight' or a 'transition condition' etc. But a simple edge with no data might be nice to have in here anyway.